lib/IMAP/ImapMessageFetcher.php

```diff
@@ -512,19 +515,29 @@ private function parseHeaders(Horde_Imap_Client_Data_Fetch $fetch): void {
 			$this->dispositionNotificationTo = $dispositionNotificationTo->value_single;
 		}
 
-		$listUnsubscribeHeader = $parsedHeaders->getHeader('list-unsubscribe');
-		if ($listUnsubscribeHeader !== null) {
-			$listHeaders = new Horde_ListHeaders();
-			/** @var Horde_ListHeaders_Base[] $headers */
-			$headers = $listHeaders->parse($listUnsubscribeHeader->name, $listUnsubscribeHeader->value_single);
-			foreach ($headers as $header) {
-				if (str_starts_with($header->url, 'http')) {
-					$this->unsubscribeUrl = $header->url;
-					break;
-				}
-				if (str_starts_with($header->url, 'mailto')) {
-					$this->unsubscribeMailto = $header->url;
-					break;
+		$dkimSignatureHeader = $parsedHeaders->getHeader('dkim-signature');
```

Comment:
**nit:** early return would lead to better readability here and save about 3 brackets :wink:

Reply:
Thanks. I will take that into account for the follow-up to add the DKIM signature validation.